src/page/index.js

```diff
@@ -19,36 +21,165 @@ import {
-const newProfileModal = new PopupWithForm("#edit-profile-modal", (values) => {
-  userInformation.setUserInfo(values);
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "79fb9209-ee9a-4da9-a138-5a38df01e74c",
+    "Content-Type": "application/json",
+  },
 });
+const profileImage = document.querySelector(".profile__image");
+
+const userInformation = new UserInfo(
+  profileTitle,
+  descriptionJob,
+  profileImage
+);
+const newProfileModal = new PopupWithForm(
+  "#edit-profile-modal",
+  handleProfileFormSubmit
+);
 newProfileModal.setEventListeners();
+
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  // here we change the button text
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      // We need to close only in `then`
+      popupInstance.close();
+    })
+    // we need to catch possible errors
+    // console.error is used to handle errors if you don’t have any other ways for that
+    .catch(console.error)
+    // in `finally` we need to return the initial button text back in any case
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
+
+// here is an example of the profile form handling
+function handleProfileFormSubmit(inputValues) {
+  // we create a function that returns a promise
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+    return api
+      .editUserInfo(inputValues)
+      .then((userData) => {
+        userInformation.setUserInfo(userData);
+      })
+      .catch(console.error);
+  }
+  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
+  handleSubmit(makeRequest, newProfileModal);
+}
+
 const imagePreview = new PopupWithImage(".modal-images-preview");
 imagePreview.setEventListeners();
 
+const profileAvatarEdit = new PopupWithForm(
+  "#edit-profile-image",
+  handleProfileAvatar
+);
+profileAvatarEdit.setEventListeners();
+
+api.getUserInformation().then((user) => {
+  userInformation.setUserInfo({ name: user.name, about: user.about });
+  userInformation.setAvatar(user.avatar);
+});
+
 function handleCardFormSubmit(data) {
-  const title = data.title;
-  const image = data.link;
-  const card = createCard({
-    name: title,
-    link: image,
+  const cardInfo = {
+    name: data.title,
+    link: data.link,
+  };
+  newCardModal.renderLoading(true);
+  api
+    .createCard(cardInfo)
+    .then((res) => {
+      const card = createCard({
+        name: res.name,
+        link: res.link,
+        _id: res._id,
+        isLiked: res.isLiked,
+      });
+      section.addItem(card);
+      newCardModal.close();
+      formValidators["add-card-form"].resetValidation();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      newCardModal.renderLoading(false);
```

Comment:
**Could be improved**

Try to use `handleSubmit` for any modal with a form submitting